extensions/amp-fx-collection/0.1/scroll-toggle.js

```diff
@@ -0,0 +1,282 @@
+  onScroll_(scrollTop) {
+    this.lastScrollTop_ = scrollTop;
+
+    const delta = this.lastScrollTop_ - this.baseScrollTop_;
+
+    // If we are scrolling in the direction that doesn't change our
+    // header status, update the base scroll position.
+    if ((!this.isShown_ && delta > 0) ||
+        (this.isShown_ && delta < 0)) {
+      this.baseScrollTop_ = this.lastScrollTop_;
+    }
+
+    if (!this.isShown_ &&
+        this.lastScrollTop_ <= HEIGHT_PX) {
+      // If we reached the top, then we need to immediately animate
+      // the target back in irrespective of our current animation
+      // state.
+      this.toggle_(true);
+      this.baseScrollTop_ = this.lastScrollTop_;
+      return;
+    }
+    if (!this.isShown_ &&
+        delta < -RETURN_THRESHOLD_PX) {
+      // If we scrolled up enough up to show the target again, do so.
+      this.toggle_(true);
+      this.baseScrollTop_ = this.lastScrollTop_;
+      return;
+    }
+    if (this.isShown_ &&
+        delta > ANIMATION_THRESHOLD_PX) {
+      // If we scrolled down enough to animate the target out, do so.
+      this.toggle_(false);
+      this.baseScrollTop_ = this.lastScrollTop_;
+    }
+  }
+
+  /**
+   * @param {boolean} isShown
+   * @private
+   */
+  toggle_(isShown) {
+    if (this.isShown_ == isShown) {
+      return;
+    }
+    this.isShown_ = isShown;
+    this.observable_.fire(isShown);
+  }
+}
+
+/**
+ * MUST be done inside mutate phase.
+ * @param {!Element} element
+ * @param {number} offset
+ */
+export function scrollToggleFloatIn(element, offset) {
+  setImportantStyles(element, {
+    'transition': `transform ${ANIMATION_DURATION_MS}ms ${ANIMATION_CURVE}`,
+    'transform': `translate(0, ${px(offset)})`,
+  });
+}
+
+/**
+ * MUST be done inside measure phase.
+ * @param {!Element} element
+ * @param {boolean} isShown
+ * @param {!ScrollTogglePosition} position
+ * @return {number}
+ */
+export function getScrollToggleFloatInOffset(element, isShown, position) {
+  if (isShown) {
+    return 0;
+  }
+  const offset = element./*OK*/getBoundingClientRect().height;
+  if (position == ScrollTogglePosition.TOP) {
+    return -offset;
+  }
+  return offset;
+}
+
+
+/**
+ * @param {!Element} element
+ * @param {!Object<string, string>} computedStyle
+ * @return {boolean}
+ */
+export function assertValidScrollToggleElement(element, computedStyle) {
+  return (
+    assertStyleOrWarn(computedStyle, 'overflow', 'hidden', element) &&
+    assertStyleOrWarn(computedStyle, 'position', 'fixed', element));
+}
+
+/**
+ * @param {!Element} element
+ * @param {string} type
+ * @param {!Object<string, string>} computedStyle
+ * @return {?ScrollTogglePosition}
+ */
+export function getScrollTogglePosition(element, type, computedStyle) {
+  const position = type.replace(/^float\-in\-([^\s]+)$/, '$1');
+
+  devAssert(position.length > 0);
+
+  // naming convention win:
+  // position `top` should have `top: 0` and `bottom` should have `bottom: 0`
+  if (!assertStyleOrWarn(
```

Comment:
I don't think this works in the viewer case. Top for all position fixed items are updated by Fixed layer as the viewer header moves around. even if it works, there is probably a race between this running and fixed elements getting updated.

Let's test the sample in the new viewer sandbox.

Reply:
Super broken, disabled viewer for now.